bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataClientWrapper.java

```diff
@@ -46,39 +60,112 @@ public BigtableDataClientWrapper(BigtableDataClient bigtableDataClient,
-    throw new UnsupportedOperationException("Not implemented yet");
+  public ListenableFuture<Row> readModifyWriteRowAsync(ReadModifyWriteRow readModifyWriteRow) {
+    ListenableFuture<ReadModifyWriteRowResponse> response =
+        delegate.readModifyWriteRowAsync(readModifyWriteRow.toProto(requestContext));
+    return Futures.transform(response, new Function<ReadModifyWriteRowResponse, Row>() {
+      @NullableDecl
+      @Override
+      public Row apply(@NullableDecl ReadModifyWriteRowResponse input) {
+        return convert(input.getRow());
+      }
+    }, MoreExecutors.directExecutor());
   }
 
+  /** {@inheritDoc} */
   @Override
   public IBulkMutation createBulkMutationBatcher() {
     throw new UnsupportedOperationException("Not implemented yet");
   }
 
+  /** {@inheritDoc} */
   @Override
-  public ApiFuture<Boolean> checkAndMutateRowAsync(ConditionalRowMutation conditionalRowMutation) {
-    throw new UnsupportedOperationException("Not implemented yet");
+  public ListenableFuture<Boolean> checkAndMutateRowAsync(
+      ConditionalRowMutation conditionalRowMutation) {
+    final CheckAndMutateRowRequest request = conditionalRowMutation.toProto(requestContext);
+    final ListenableFuture<CheckAndMutateRowResponse> response =
+        delegate.checkAndMutateRowAsync(request);
+    return Futures.transform(response, new Function<CheckAndMutateRowResponse, Boolean>() {
+      @NullableDecl
+      @Override
+      public Boolean apply(@NullableDecl CheckAndMutateRowResponse input) {
+        return wasMutationApplied(request, input);
+      }
+    }, MoreExecutors.directExecutor());
   }
 
+  /** {@inheritDoc} */
   @Override
   public Boolean checkAndMutateRow(ConditionalRowMutation conditionalRowMutation) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    CheckAndMutateRowRequest request = conditionalRowMutation.toProto(requestContext);
+    CheckAndMutateRowResponse response =
+        delegate.checkAndMutateRow(conditionalRowMutation.toProto(requestContext));
+    return wasMutationApplied(request, response);
+  }
+
+  /**
+   * This method is referred from CheckAndMutateUtil#wasMutationApplied.
+   */
+  private static boolean wasMutationApplied(CheckAndMutateRowRequest request,
+      CheckAndMutateRowResponse response) {
+
+    // If we have true mods, we want the predicate to match.
+    // If we have false mods, we do not want the predicate to match.
+    return (request.getTrueMutationsCount() > 0
+        && response.getPredicateMatched()) || (
+        request.getFalseMutationsCount() > 0
+            && !response.getPredicateMatched());
+  }
+
+  /**
+   * This method converts instances of {@link com.google.bigtable.v2.Row} to {@link Row}.
+   *
+   * @param row an instance of {@link com.google.bigtable.v2.Row} type.
+   * @return {@link Row} an instance of {@link Row}.
+   */
+  private static Row convert(com.google.bigtable.v2.Row row) {
```

Comment:
Please name this method a bit more specific

Reply:
update method name to `transformResponse()`